src/java/org/apache/cassandra/cql3/Tuples.java

```diff
@@ -51,11 +44,30 @@ public static ColumnSpecification componentSpecOf(ColumnSpecification column, in
                                        (getTupleType(column.type)).type(component));
     }
 
+    public static ColumnSpecification makeReceiver(List<? extends ColumnSpecification> receivers)
```

Comment:
Sorry, I do not really understand you concern. We only have a single class to represent markers.

Reply:
The approach is now different. We have a single Marker class for a single type because we use the TupleType instead of using the the type of its elements as it was done in Tuples before.